screens/ClerkLoginScreen.js

```diff
@@ -99,48 +70,59 @@ export default class ClerkLogin extends React.Component {
             }
           }
         );
+    }).catch(err => {
+      console.error("Error looking up clerk", err);
     });
   }
 
   // From SignUpScreen. Sign in function. It sets the user token in local storage
   // to be the fname + lname and then navigates to homescreen.
   _asyncSignin = async recordId => {
     await AsyncStorage.setItem("clerkId", recordId);
+    await AsyncStorage.setItem("storeId", this.state.storeId);
     this.props.navigation.navigate("CustomerPhoneNumberScreen");
     // this.props.navigation.navigate("ClerkLoginScreen");
   };
 
   // This function will sign the user in if the clerk is found.
   async handleSubmit() {
+    await BASE("Stores")
+      .find(this.state.storeId)
+      .then(storeRecord => {
+        if (storeRecord) {
+          let name = storeRecord["fields"]["Store Name"];
+          this.setState({ storeName: name });
+        }
+      });
     await this.lookupClerk(this.state.storeName, this.state.password)
       .then(resp => {
         if (resp) {
           const recordId = resp;
-          this.setState({
-            storeName: stores[0],
-            password: ""
-          });
           this._asyncSignin(recordId);
         }
       })
       .catch(err => {
         console.log(err);
-        this.setState({ storeName: stores[0], password: "" });
+        this.setState({ storeName: this.state.stores[0], password: "" });
```

Comment:
What I learned from Franco: 
There are two main ways to do promises - 1) promise chaining .then()->  .catch; 2) async/await + try/catch block

For #1 the function that is invoked, for example `lookupCustomer`, returns a promise; because it returns a promise, we can just invoke the function and use `.then().catch()` to process the resolved/rejected promises. If you have multiple functions to `await`, you can do it all in one big `try/catch` block; the `catch` will catch first error thrown!

Thus, probably shouldn't do this in a `.catch` block; it wouldn't be invoked unless an error is thrown. We should call `setState` in the `.then` of the `_asyncSignin` call - but might be better to `await` it in the try-catch block along with the Airtable API call.

Reply:
Oops, we shouldn't be changing state at that point. When we switch the screen we don't need to reset the form anymore. Was leftover from testing. We should just be console erroring in the catch block (is that ok?)

Reply:
Yep! Sounds good